### ISSUE TITLE
Support multiple databases connections with different crypto keys

### ISSUE THREAD

**Issue by peterfarrell** [incuna/django-pgcrypto-fields#67]
Add support for multiple databases that have different crypto keys. Just move PUBLIC_PGP_KEY, PRIVATE_PGP_KEY, and PGCRYPTO_KEY into the DATABASES configuration and grab that information from the connection instead of reading from the settings:

```
DATABASES = {
    'default': {
        'ENGINE': 'django.db.backends.postgresql_psycopg2',
        'PUBLIC_PGP_KEY': open(PUBLIC_PGP_KEY_PATH, 'r').read(),
        'PRIVATE_PGP_KEY': open(PRIVATE_PGP_KEY_PATH, 'r').read(),
        'PGCRYPTO_KEY': 'super_secret_key',
        ...
    },
}
```

DecryptedCol
```python
    def as_sql(self, compiler, connection):
        sql, params = super(DecryptedCol, self).as_sql(compiler, connection)
        decrypt_sql = self.decrypt_sql_template.format(
            key=self.target._get_encryption_key(connection),
            dbtype=self.target._get_base_db_type(connection),
            sql=sql
        )
        return decrypt_sql, params
```

This is easily made backwards compatible:

```python
   def _get_encryption_key(self, connection):
        if 'PGCRYPTO_KEY' in connection.settings_dict:
            key = connection.settings_dict['PGCRYPTO_KEY']
        else:
            key = getattr(settings, 'PGCRYPTO_KEY', settings.SECRET_KEY)
        # Escape any percent symbols in the key, to avoid them being
        # interpreted as extra substitution placeholders later on.
        key = key.replace('%', '%%')
        return key
```

**Comment by meshy** [comment]
Perhaps it would make sense for these settings to be in the [OPTIONS](https://docs.djangoproject.com/en/2.1/ref/settings/#std:setting-OPTIONS) part of the database setting.

**Comment by peterfarrell** [comment]
@meshy I've thought adding them to the `OPTION` dict but that dict is passed to the database -- not sure if that is done over the wire in clear text or not. Seems like possibility to have key conflicts with any future postgresql options or possibly security issues. Adding them to the root dict for the database connection is less risky security wise.

**Comment by meshy** [comment]
@peterfarrell OK, sounds good!

Thanks for all the hard work you're putting into this library :)

**Comment by peterfarrell** [comment]
This has been merged into Master for the 2.5.0 release.
